### ISSUE TITLE
Use `fieldset` for tabledrag components

### ISSUE THREAD

**Issue by lkmorlan** [bcgov/MFIN-Data-Catalogue#277]
**OP timer**

https://openplus.monday.com/boards/4092908516/pulses/5682991850

---

## Describe the bug

On tabledrag components, such as "Data sets used" on build section 3, the label for the component should be the `legend` for a `fieldset` which wraps the entire component, including the "Add another item" button. This will help with accessibility.

This can be implemented by overriding `bootstrap5/templates/form/field-multiple-value-form.html.twig`.
